syne_tune/optimizer/schedulers/hyperband_pasha.py

```diff
@@ -81,19 +81,21 @@ def _get_top_two_rungs_rankings(self):
             self._rungs[-self.current_rung_idx],
             self._rungs[-self.current_rung_idx + 1],
         ]:
-            if rung.data != {}:
-                trial_ids = rung.data.keys()
-                values = []
-                for trial_id in trial_ids:
-                    values.append(rung.data[trial_id][0])
-                # order specifies where the value should be placed in the sorted list
-                values_order = np.array(values).argsort()
-                # calling argsort on the order will give us the ranking
-                values_ranking = values_order.argsort()
+            if len(rung) > 0:
+                # Note that entries in ``rung.data`` are already sorted
+                trial_ids, values = zip(
+                    *[(x.trial_id, x.metric_val) for x in rung.data]
+                )
+                # Note: To retain the exact logic of older code (which used
+                # ``numpy.argsort``, we assume the ranks for increasing metric
+                # values, which is the opposite ordering to ``rung.data`` if
+                # ``self._mode == "max"``
+                if self._mode == "min":
+                    values_ranking = list(range(len(trial_ids)))
+                else:
+                    values_ranking = list(range(len(trial_ids) - 1, -1, -1))
```

Comment:
most likely this entire function could be dropped now given that things are sorted already